src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.test.tsx

```diff
@@ -202,4 +216,33 @@ describe('Activity Flow Builder', () => {
       });
     });
   });
+
+  test('Ensures no reviewable activities remain when adding/replacing an activity)', async () => {
+    renderActivityFlowBuilder(mockedAppletFormDataWithReviewableActivity);
+
+    const addActivity = screen.getByTestId(`${mockedActivityFlowBuilderTestid}-add`);
+    await userEvent.click(addActivity);
+
+    const activitiesInAddActivity = screen.getAllByTestId(
+      new RegExp(`^${mockedActivityFlowBuilderTestid}-add-activity-\\d+$`),
```

Comment:
Let's move `new RegExp(`^${mockedActivityFlowBuilderTestid}-add-activity-\\d+$`)` into a constant here and in row 243 to reduce new object creation in the loops.